internal/catalogmetadata/filter/filter.go

```diff
@@ -0,0 +1,51 @@
+package filter
+
+import (
+	"github.com/operator-framework/operator-controller/internal/catalogmetadata"
+)
+
+// Predicate returns true if the object should be kept when filtering
+type Predicate[T catalogmetadata.Schemas] func(entity *T) bool
+
+// Filter filters a slice accordingly to
+func Filter[T catalogmetadata.Schemas](in []*T, test Predicate[T]) []*T {
+	out := []*T{}
+	for i := range in {
+		if test(in[i]) {
+			out = append(out, in[i])
+		}
+	}
+	return out
+}
+
+func And[T catalogmetadata.Schemas](predicates ...Predicate[T]) Predicate[T] {
+	return func(obj *T) bool {
+		eval := true
+		for _, predicate := range predicates {
+			eval = eval && predicate(obj)
+			if !eval {
+				return false
+			}
+		}
+		return eval
```

Comment:
Pretty nitpicky but could we not just say:
```suggestion
		for _, predicate := range predicates {
			if !predicate(obj) {
				return false
			}
		}
		return true
```

Reply:
Good point! I just copy-pasted this from older [deppy commit](https://github.com/operator-framework/deppy/blob/4c15c99c52e42655253fcfac942d805d46cd8d22/pkg/deppy/input/query.go#L32-L56) without paying too much attention.

I'll update.